src/ServiceAccount/Search/FilterSelect.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  Select,
+  SelectOption,
+  SelectOptionObject,
+  SelectVariant,
+  ToolbarItem,
+} from '@patternfly/react-core';
+
+export type FilterSelectProps = {
+  filterSelected?: string;
+  setFilterSelected: (value: string) => void;
+};
+
+export const FilterSelect: React.FunctionComponent<FilterSelectProps> = ({
+  setFilterSelected,
+  filterSelected,
+}) => {
+  const { t } = useTranslation();
+  const [isFilterExpanded, setIsFilterExpanded] = useState(false);
+
+  const onToggle = () => {
+    setIsFilterExpanded(!isFilterExpanded);
+  };
+
+  const onSelect = (
+    _event: React.MouseEvent | React.ChangeEvent,
+    selection: string | SelectOptionObject
+  ) => {
+    setIsFilterExpanded(!isFilterExpanded);
+    setFilterSelected(selection?.toString());
+  };
```

Comment:
```suggestion
  const onSelect = useCallback((
      _event: React.MouseEvent | React.ChangeEvent,
      selection: string | SelectOptionObject
    ) => {
      if (selection && typeof selection === 'string) {
        setIsFilterExpanded(false);
        onChange(selection);
      }
    },
    [onChange]
  );
```

Reply:
Line 31: the type from the PF select for `selection` isn't great, so we need to do some checks on its value to let Tyepscript help us.

Line 32: you always want to close the dropdown when something gets selected. If you toggle the value of `isFilterExpanded`, you risk setting it `true` if something else for some weird reason changes it (which should not happen, but better safe than sorry).

Line 27 and 36: this needs to be wrapped in `useCallback`. The function depends on `onChange`, that's a property that is passed to the `FilterSelect` component and might change between renders. So we add it to the list of dependencies for `useCallback`